gelato-relay/src/types.rs

```diff
@@ -0,0 +1,130 @@
+use serde::{Deserialize, Serialize};
+
+/// Request for forwarding tx to gas-tank based relay service
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct ForwardRequest {
```

Comment:
Why doesn't `UnsignedFowardRequest` live here? Also, check the `enum` comment from there as it should be applied here.

Reply:
Wanted to keep `rust/gelato-relay` minimal so just methods for making gelato requests + corresponding structs that get serialized in http request. 

`UnsignedForwardRequest` implements EIP712, which is a standard for signing data structures. Didn't want to add this impl on the gelato-relay `ForwardRequest` since it would be adding a lot of extra functionality to type that should just be a web request struct that gets serialized.

Reply:
Basically `gelato-relay` should have no knowledge of what EIP 712 is

Reply:
Also realized they have one field that's different, `type_id`

Reply:
I somewhat disagree. If we're building a rust SDK for Gelato, we should leverage the type system to help users use it correctly. The implementation as is has a ton of sharp edges and poor typing (I assume inherited from the JS/TS it was based on)

I think it should have knowledge of 712, if 712 is required to use Gelato correctly